src/sisl_toolbox/siesta/minimizer/_minimize.py

```diff
@@ -332,6 +343,77 @@ def run(self, *args, **kwargs):
         return _convert_optimize_result(self, opt)
 
 
+@set_module("sisl_toolbox.siesta.minimizer")
+class BADSMinimize(BaseMinimize):
+    """Bayesian Adaptive Direct Search (BADS) minimizer.
+
+    It uses the pybads package to perform the minimization.
+    """
+
+    def run(self, options={}, get_hard_bounds: Callable = lambda x: x, **kwargs):
```

Comment:
I would rename `get_hard_bounds` to `func_conv_bounds_to_hard` or something more descriptive.

Reply:
Whatever you prefer